File: app/src/main/java/ch/bailu/aat/services/location/LocationStackItem.java
package ch.bailu.aat.services.location;

import android.content.Context;

import java.io.Closeable;

import ch.bailu.aat.preferences.PresetDependent;

public abstract class LocationStackItem implements Closeable, PresetDependent{
    public abstract void passState(int state);
    public abstract void passLocation(LocationInformation location);


    @Override
    public void close(){}


    @Override
    public void preferencesChanged(Context c, int i) {}


    public void appendStatusText(StringBuilder builder) {
        builder.append("<b>");
        builder.append(getClass().getSimpleName());
        builder.append("</b><br>");
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/service/location/LocationStackItem.kt
package ch.bailu.aat_lib.service.location

import ch.bailu.aat_lib.preferences.OnPresetPreferencesChanged
import ch.bailu.aat_lib.preferences.StorageInterface
import java.io.Closeable
import javax.annotation.Nonnull

abstract class LocationStackItem : Closeable, OnPresetPreferencesChanged {
    abstract fun passState(state: Int)
    abstract fun passLocation(location: LocationInformation)

    override fun close() {}

    override fun onPreferencesChanged(storage: StorageInterface, key: String, presetIndex: Int) {}

    open fun appendStatusText(builder: StringBuilder) {
        builder.append("<b>")
        builder.append(javaClass.simpleName)
        builder.append("</b><br>")
    }
}
